planex/mock.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/python
+
+"""
+planex-mock: Wrapper around mock
+"""
+
+import subprocess
+import sys
+from uuid import uuid4
+
+import argparse
+import argcomplete
+from planex.util import add_common_parser_options
+
+
+def parse_args_or_exit(argv):
+    """
+    Parse command line options
+    """
+    parser = argparse.ArgumentParser(
+        description='Planex build system in a chroot (a mock wrapper)')
+    add_common_parser_options(parser)
+    parser.add_argument(
+        "--configdir", metavar="CONFIGDIR", default=None,
+        help="Change where the config files are found")
+    parser.add_argument(
+        "--resultdir", metavar="RESULTDIR", default=None,
+        help="Path for resulting files to be put")
+    parser.add_argument(
+        "-D", "--define", default=[], action="append",
+        help="--define='MACRO EXPR' \
+              define MACRO with value EXPR for the build")
+    argcomplete.autocomplete(parser)
+    return parser.parse_known_args(argv)
+
+
+def extract_srpms(passthrouhg_args):
+    """
+    Filter the list of passthrouhg arguments to check if they are all
+    look like SRPMs. Return a tuple with a list of the SRPMs and
+    a list of the other unparsed arguments.
+    """
+    srpms, errors = [], []
+
+    for arg in passthrouhg_args:
+        if arg.endswith(".src.rpm") or arg.endswith(".srpm"):
+            srpms.append(arg)
+        else:
+            errors.append(arg)
+
+    return srpms, errors
+
+
+def get_command_line(intercepted_args, defaults, srpms):
+    """
+    Return mock command line and arguments
+    """
+    cmd = ['mock']
+    if intercepted_args.quiet:
+        cmd.append('--quiet')
+    for define in intercepted_args.define:
+        cmd.append('--define')
+        cmd.append(define)
+    if intercepted_args.configdir is not None:
+        cmd.append('--configdir')
+        cmd.append(intercepted_args.configdir)
+    if intercepted_args.resultdir is not None:
+        cmd.append("--resultdir")
+        cmd.append(intercepted_args.resultdir)
+    cmd.extend(defaults)
+    for srpm in srpms:
+        cmd.append(srpm)
+
+    return cmd
+
+
+def main(argv):
+    """
+    Entry point arguments: srpm0 srpm1 srpm2 ...
+    At least one SRPM file (with ".src.rpm" or ".srpm" extension)
+    should be present.
+    """
+
+    if len(sys.argv) == 0:
```

Comment:
I think `argv[0]` is the name of the binary and hence, you need `len(sys.argv) <= 1`. Update: I noticed below that you are passing the right list of arguments but now I think the name here should not be `argv` to avoid the confusion (unless this is a Python convention).